js/node/src/inference_session_wrap.cc

```diff
@@ -8,11 +8,15 @@
 #include "run_options_helper.h"
 #include "session_options_helper.h"
 #include "tensor_helper.h"
+#include "directml_load_helper.h"
 
 Napi::FunctionReference InferenceSessionWrap::constructor;
 Ort::Env *InferenceSessionWrap::ortEnv;
 
 Napi::Object InferenceSessionWrap::Init(Napi::Env env, Napi::Object exports) {
+#ifdef _WIN32
+  LoadDirectMLDll(env);
```

Comment:
We can also wrap this in a `#ifdef USE_DML`, since we don't need to load DML if ORT wasn't built for it.

Reply:
I can wrap it, but from my understanding, as this particular project already has binaries for all platforms bundled, it should always have a build with DML enabled on Windows. Even on ARM? I see DML supports ARM platform but I haven't tested it. Also Linux should be compiled with CUDA support and Apple with CoreML (it already is).

However, from my tests enabling CoreML provider did not give any performance benefits, but i guess i should add code for it. Maybe it can be beneficial for small models. And maybe for TensorRT, but last time i tried it, it refused to load models with inputs with variable shapes. Not very useful.

Anyway, realistically, it should never fail. As it will be bundled with binaries compiled with DML support and DirectML.dll
But it will throw an error later, when user creates a session with "dml" provider from JS code if GPU is not supported

Reply:
> I see DML supports ARM platform but I haven't tested it.

Yep, it runs fine on my teammate's Surface Pro X.

For CoreML and TensorRT opinions, I defer to others.

Reply:
Actually, it looks like i need to wrap CUDA and TensorRT code with #ifdef USE_CUDA/USE_TENSORRT. Because there are no CUDA libraries for Windows ARM

Reply:
Normally this inference_session_wrap.cc is built with the other parts of ONNX Runtime at the same time. If the other files are built with USE_DML, this one should be too. So, adding a "#if defined(USE_DML)" here would not be harmful.